### ISSUE TITLE
BatchNorm behavior

### ISSUE THREAD

**Issue by EmanueleGhelfi** [tensorflow/tensorflow#29480]
**Describe the feature and the current behavior/state.**
Currently the BatchNorm layer behavior is the following (if I have correctly understood):
If `training=True`:
- The output is centered and scaled using the moving mean and moving variance are updated with the current minibatch mean and variance. The moving mean and variance are updated.

If `training=False`:
- The output is centered and scaled using the moving mean and moving variance. The current batch statistics are not updated. 

I think this is a problem in some cases. For example in pix2pix (code: https://www.tensorflow.org/alpha/tutorials/generative/pix2pix) (paper: https://arxiv.org/abs/1611.07004) the evaluation is done using the flag `training=True` in order to use also the current batch statistics (apart for the dropout layer).

I think that it would be useful to add an additional flag telling whether to use the current batch statistics without updating the moving mean and moving variance. In this case it will be possible to evaluate correctly models like pix2pix. 

In other words: 
The batch norm call can be changed into:
`keras.layers.BatchNormalization()(input, training, update_statistics)`

With the current implementation is very difficult to evaluate models in which we need to use the flag `training=True`without updating the statistics.
